docs/user/huggingface.rst

```diff
@@ -63,6 +66,63 @@ To install accelerate_, run the following command inside your Python environment
     :class:`torch.optim.LBFGS`), you cannot use accelerate.
 
 
+Caution when using a multi-GPU setup
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+There is a known issue when using accelerate in a multi-GPU setup *if copies of
+the net are created*. In particular, be aware that sklearn often creates copies
+under the hood, which may not immediately obvious to a user. Examples of
+functions and classes creating copies are:
+
+- `GridSearchCV`, `RandomizedSearchCV` etc.
+- `cross_validate`, `cross_val_score` etc.
+- `VotingClassifier`, `CalibratedClassifierCV` and other meta estimators (but
+  not `Pipeline`).
+
+When using any of those in a multi-GPU setup with :class:`.AccelerateMixin`, you
+may encounter errors. A possible fix is to prevent the ``Accelerator`` instance
+from being copied (or, to be precise, deep-copied):
+
+.. code:: python
+
+    class AcceleratedNet(AccelerateMixin, NeuralNet):
+        pass
+
+    class MyAccelerator(Accelerator):
+        def __deepcopy__(self, memo):
+            return self
```

Comment:
I'm trying to come up with way this can break, but it seems okay. From my understanding, here is how I see the behavior:

- `GridSearchCV.fit` will `clone` the `NeuralNet`, and the PyTorch Module gets deep copied and leave the `Accelerator` in place.
- During `initialize`, the Accelerator will be reconfigured to point to the new Module (If there was a way to "reset the Accelerator" that would be nice too)
- Profit?

Maybe passing an stateful Accelerator object into `AccelerateMixin` was a bad design choice. For `NeutralNet`, I usually prefer passing in the Module class and not an instance. For `AccelerateMixin`, we can allow `accelerator__mixed_precision="fp16"`.

Reply:
> From my understanding, here is how I see the behavior: [...]

Sorry, I don't quite understand, is this behavior without the `__deepcopy__` patch or with?

> * If there was a way to "reset the Accelerator" that would be nice too

AFAIK, there is no public way to do that (see [here](https://github.com/skorch-dev/skorch/blob/c909e14a1d66de1de23a4619a3c3dc0ef5498a44/skorch/tests/test_hf.py#L534)). The `Accelerator` was always intended to be run in a script that has a single training loop, re-using is normally out of scope. Devs told me there should only be one instance, and it should be created as early as possible in the script.

> Maybe passing an stateful Accelerator object into `AccelerateMixin` was a bad design choice.

Yes, possibly the pattern you suggest should have been the default, although it would not solve the issue with there only ever being a single `Accelerator` instance. In general, though, I believe we could add the suggested pattern without BC breaking, allowing users to pass either instances or classes like with `module`.

Reply:
Got it. I created an issue, #960, for tracking what we discussed about instantiating the `Accelerator`.